app/controllers/redirections_controller.rb

```diff
@@ -1,4 +1,6 @@
 class RedirectionsController < ApplicationController
+  before_action :ensure_referrer_is_not_localhost
```

Comment:
Good points. I've used rescue_from to great effect before, but this is probably not the place for it.

What about:

```ruby
class Redirection < ActiveRecord::Base
  validates :url, format: { with: -> /regex for no localhost/ }
end
```

and then redirect if it doesn't validate with that format?